django/contrib/auth/tokens.py

```diff
@@ -50,7 +50,11 @@ def check_token(self, user, token):
             return False
 
         # Check that the timestamp/uid has not been tampered with
-        if not constant_time_compare(self._make_token_with_timestamp(user, ts), token):
+        attempts = [
```

Comment:
Same as below, I think I'd prefer a loop that breaks early.

Reply:
I've gone with this: 

```diff
         # Check that the timestamp/uid has not been tampered with
-        attempts = [
-            constant_time_compare(self._make_token_with_timestamp(user, ts, secret), token)
-            for secret in self.secrets
-        ]
-        if not any(attempts):
+        valid_compare = False
+        for secret in self.secrets:
+            valid_compare = constant_time_compare(self._make_token_with_timestamp(user, ts, secret), token)
+            if valid_compare:
+                break
+        if not valid_compare:
             return False
```

(Incoming). Happy to take style points on that. 😄 (But I think it's what you meant)

Reply:
What about:
```py
for secret in self.secrets:
    if constant_time_compare(self._make_token_with_timestamp(user, ts, secret), token):
        break
    else:
        return False        
```

Reply:
@florian. Yes! That's nicer. (incoming.) 👍